src/proxy-animation.js

```diff
@@ -780,7 +780,7 @@ function createProxyEffect(details) {
   const updateTimingHandler = {
     apply: function(target, thisArg, argumentsList) {
       // Additional validation that is specific to scroll timelines.
-      if (details.timeline) {
+      if (details.timeline && argumentsList[0]) {
```

Comment:
I see that further down you have this check:

```js
if (typeof duration !== 'undefined' && duration !== 'auto') {
```

This check for `undefined` can be omitted then I guess?

---

Furthermore, wouldn’t it be possible to do this check for `argumentsList[0]` earlier, and have it exit prematurely if `argumentsList.length` equals `0`?

I.e.
```js
apply: function(target, thisArg, argumentsList) {
  if (!argumentsList || !argumentsList.length) return;

  if (details.timeline) {
  …
```

Reply:
> I see that further down you have this check:
> 
> ```js
> if (typeof duration !== 'undefined' && duration !== 'auto') {
> ```
> 
> This check for `undefined` can be omitted then I guess?
> 

I don't think we can omit that check. 

`updateTiming()` is used to perform a delta update to the timing, and the properties of the [OptionalEffectTiming](https://drafts.csswg.org/web-animations-1/#dictdef-optionaleffecttiming) argument don't have to be set.

We should only remove the autoDurationEffect flag if the duration property is set to a new value.

```js
effect.updateTiming({ iterations: 2 }); 
// Duration is undefined, and we should not clear the autoDuration flag.

effect.updateTiming({ duration: 200 });
// Duration is updated to a non-auto value, and we need to clear the flag.
```

We could clear the flag even if the duration is 'auto', as it will be set again when we renormalize time: 
```js
        if (typeof duration !== 'undefined') {
          details.autoDurationEffect = null
        }
```

I could open a separate PR to simplify that check?

On a side note, maybe we need better naming for the flag?
The flag is used to keep track of whether the user specified an auto-duration, as the 'auto' value will disappear when we normalize the time to INTERNAL_DURATION_MS in `getTiming()`.

> Furthermore, wouldn’t it be possible to do this check for `argumentsList[0]` earlier, and have it exit prematurely if `argumentsList.length` equals `0`?
> 
> I.e.
> 
> ```js
> apply: function(target, thisArg, argumentsList) {
>   if (!argumentsList || !argumentsList.length) return;
> 
>   if (details.timeline) {
>   …
> ```

Absolutely. I've updated the PR to do this instead.

Reply:
I think this looks good for now. The mentioned refactor can be part of a separate PR.